File: example/periphery/capture/test_capture.cc
#include <lockzhiner_vision_module/edit/edit.h>

#include <iostream>
#include <opencv2/core/core.hpp>
#include <opencv2/highgui/highgui.hpp>
#include <opencv2/imgproc/imgproc.hpp>

int main(int argc, char *argv[]) {
  int width = 640;
  int height = 480;
  if (argc == 3) {
    width = atoi(argv[1]);
    height = atoi(argv[2]);
  }

  lockzhiner_vision_module::edit::Edit edit;
  // 打开服务器并阻塞接收客户端连接
  if (!edit.StartAndAcceptConnection()) {
    std::cout << "Failed to start and accept connection" << std::endl;
    return 1;
  }
  std::cout << "Succeed to connect device" << std::endl;

  cv::VideoCapture cap;
  // 设置摄像头长宽
  cap.set(cv::CAP_PROP_FRAME_WIDTH, width);
  cap.set(cv::CAP_PROP_FRAME_HEIGHT, height);
  // 打开摄像头
  cap.open(0);

  const int w = cap.get(cv::CAP_PROP_FRAME_WIDTH);
  const int h = cap.get(cv::CAP_PROP_FRAME_HEIGHT);
  cv::Mat temp_mat;
  while (true) {
    cap >> temp_mat;
    // 判断获取的图片是否为空
    if (temp_mat.empty()) {
      continue;
    }
    // 向客户端输出图片
    if (!edit.Print(temp_mat)) {
      std::cout << "Failed to print to edit" << std::endl;
      break;
    }
  }
  return 0;
}

File: example/periphery/capture/python/test_capture.py
from lockzhiner_vision_module.cv2 import VideoCapture
from lockzhiner_vision_module.edit import Edit
import time
import sys

if __name__ == "__main__":
    args = sys.argv
    if len(args) != 1 and len(args) != 3:
        print("Need model path. Example: python test_capture.py <width> <height>")
        exit(1)

    width = 640
    height = 480

    if len(args) == 3:
        width = int(args[1])
        height = int(args[2])
    print(f"Capture width: {width}, height: {height}")

    edit = Edit()
    edit.start_and_accept_connection()

    video_capture = VideoCapture()
    video_capture.set_width(width)
    video_capture.set_height(height)

    if video_capture.open(0) is False:
        print("Failed to open capture")
        exit(1)


    while True:
        read_index = 0
        total_time_ms = 0
        for i in range(30):
            start_time = time.time()
            ret, mat = video_capture.read()
            if ret is False:
                continue
            end_time = time.time()
            edit.print(mat)
            total_time_ms += end_time - start_time
            read_index += 1
        print(f"FPS is {1.0 / (total_time_ms/read_index)}")
